api/v1alpha1/application_types.go

```diff
@@ -432,15 +432,53 @@ func (a *Application) GetCommonSpec() *CommonSpec {
 	}
 }
 
+type Ingresses struct {
+	Hostname string
+	Internal bool
```

Comment:
can we have custom cert here too?

Reply:
Yes! Outside scope now, but a good candidate for future work.